source/4_tools/regina/src/regina.cpp

```diff
@@ -2,30 +2,892 @@
+    renderer::ResourceSlot wire_raster_state;
+
+    PassConstants pass_constants;
+
+    bool is_wireframe = false;
+
+    glm::vec3 eye_pos = {0.0f, 0.0f, 0.0f};
+    glm::mat4 view    = glm::mat4(1.0f);
+    glm::mat4 proj    = glm::mat4(1.0f);
+  };
+
+  ReginaContext g_regina_ctx; // NOLINT(fuchsia-statically-constructed-objects, cppcoreguidelines-avoid-non-const-global-variables)
+
+  //-------------------------------------------------------------------------
+  renderer::ResourceSlot get_object_committed_resource_of_frame(const renderer::ResourceSlot* frame)
+  {
+    auto it = rsl::find_if(rsl::cbegin(g_regina_ctx.frame_resource_data), rsl::cend(g_regina_ctx.frame_resource_data), [frame](const FrameData& data) { return *frame == data.frame_slot(); });
+
+    return it != rsl::cend(g_regina_ctx.frame_resource_data) ? it->object_committed_resource_slot() : renderer::ResourceSlot::make_invalid();
+  }
+  //-------------------------------------------------------------------------
+  renderer::ResourceSlot get_pass_committed_resource_of_frame(const renderer::ResourceSlot* frame)
+  {
+    auto it = rsl::find_if(rsl::cbegin(g_regina_ctx.frame_resource_data), rsl::cend(g_regina_ctx.frame_resource_data), [frame](const FrameData& data) { return *frame == data.frame_slot(); });
+
+    return it != rsl::cend(g_regina_ctx.frame_resource_data) ? it->pass_committed_resource_slot() : renderer::ResourceSlot::make_invalid();
+  }
+  //-------------------------------------------------------------------------
+  renderer::ResourceSlot get_active_object_constant_buffer_for_frame(s32 frame, s32 idx)
+  {
+    auto it = rsl::find_if(rsl::cbegin(g_regina_ctx.frame_resource_data), rsl::cend(g_regina_ctx.frame_resource_data), [frame](const FrameData& data) { return frame == data.frame_slot().slot_id(); });
+
+    return it != rsl::cend(g_regina_ctx.frame_resource_data) ? it->object_constant_buffer_slots()[idx] : renderer::ResourceSlot::make_invalid();
+  }
+  //-------------------------------------------------------------------------
+  renderer::ResourceSlot get_active_pass_constant_buffer_for_frame(s32 frame)
+  {
+    auto it = rsl::find_if(rsl::cbegin(g_regina_ctx.frame_resource_data), rsl::cend(g_regina_ctx.frame_resource_data), [frame](const FrameData& data) { return frame == data.frame_slot().slot_id(); });
+
+    return it != rsl::cend(g_regina_ctx.frame_resource_data) ? it->pass_constant_buffer_slot() : renderer::ResourceSlot::make_invalid();
+  }
+
+  //-------------------------------------------------------------------------
+  renderer::commands::CompileShaderCommandDesc create_compile_shader_parameters(const rsl::small_stack_string& shaderName, renderer::ShaderType shaderType, rsl::string_view filePath)
+  {
+    renderer::commands::CompileShaderCommandDesc compile_shader_command_desc;
+
+    switch(shaderType)
+    {
+      case renderer::ShaderType::VERTEX:
+        compile_shader_command_desc.shader_entry_point    = "VS"_tiny;
```

Comment:
you set the shader entry point here but also in the command itself. I suggest to only set it in one place

Reply:
also, shader_entry_point is a string view, so it's best to assign a string view. you're assigning a stack string to it which is destroyed at the end of this function. if you wouldn't also set it in the command (and making setting it here moot) you'd have UB and likely a crash